syntax = "proto3";

package lusp;

// 文件上传状态
enum PROTOLusp_UploadStatusInf {
    PROTOLUSP_UPLOAD_STATUS_IDENTIFIERS_COMPLETED = 0;   // 成功
    PROTOLUSP_UPLOAD_STATUS_IDENTIFIERS_PENDING   = 1;   // 等待上传
    PROTOLUSP_UPLOAD_STATUS_IDENTIFIERS_UPLOADING = 2;   // 上传部分
    PROTOLUSP_UPLOAD_STATUS_IDENTIFIERS_REJECTED  = 3;   // 拒绝上传
    PROTOLUSP_UPLOAD_STATUS_IDENTIFIERS_FAILED    = 4;   // 失败
    PROTOLUSP_UPLOAD_STATUS_IDENTIFIERS_UNDEFINED = 5;   // 未定义
}

// 上传文件类型
enum PROTOLusp_UploadFileTyped {
    PROTOLUSP_UPLOADTYPE_DOCUMENT = 0;    // 文档类型
    PROTOLUSP_UPLOADTYPE_IMAGE = 1;       // 图片类型
    PROTOLUSP_UPLOADTYPE_VIDEO = 2;       // 视频类型
    PROTOLUSP_UPLOADTYPE_AUDIO = 3;       // 音频类型
    PROTOLUSP_UPLOADTYPE_ARCHIVE = 4;     // 压缩包类型
    PROTOLUSP_UPLOADTYPE_CODE = 5;        // 代码文件
    PROTOLUSP_UPLOADTYPE_UNDEFINED = 6;   // 未知类型
}

// 文件存在策略
enum PROTOLusp_FileExistPolicy {
    PROTOLUSP_FILE_EXIST_POLICY_OVERWRITE = 0;   // 覆盖已存在的文件
    PROTOLUSP_FILE_EXIST_POLICY_SKIP = 1;        // 跳过已存在的文件
    PROTOLUSP_FILE_EXIST_POLICY_RENAME = 2;      // 重命名已存在的文件
    PROTOLUSP_FILE_EXIST_POLICY_UNDEFINED = 3;   // 未定义策略
}

// 同步上传文件信息结构体
message PROTOLusp_SyncUploadFileInfo {
    PROTOLusp_UploadFileTyped eUploadFileTyped = 1;      // 文件类型
    string sLanClientDevice = 2;                    // 客户端设备名称
    uint64 sSyncFileSizeValue = 3;                  // 同步文件大小
    string sFileFullNameValue = 4;                  // 同步全路径名称
    string sOnlyFileNameValue = 5;                  // 仅仅文件名
    string sFileRecordTimeValue = 6;                // 文件记录时间
    string sFileMd5ValueInfo = 7;                   // 文件MD5值
    PROTOLusp_FileExistPolicy eFileExistPolicy = 8;      // 文件存在策略
    string sAuthTokenValues = 9;                    // WAN上传时的tokenkey
    uint64 uUploadTimeStamp = 10;                   // 上传时间戳（毫秒级）
    PROTOLusp_UploadStatusInf eUploadStatusInf = 11;     // 上传状态
    string sDescriptionInfo = 12;                   // 描述信息
}

// 上传请求消息
message PROTOLusp_UploadRequest {
    PROTOLusp_SyncUploadFileInfo file_info = 1; // 上传文件信息
    // 可扩展其他字段，如批量上传、操作类型等
}

// 上传结果信息
message PROTOLusp_UploadResultInfo {
    bool success = 1;                // 是否成功
    uint64 uploaded_bytes = 2;       // 已上传字节数
    uint64 total_bytes = 3;          // 总字节数
    string message = 4;              // 结果描述或错误信息
    int32 error_code = 5;            // 错误码（0为成功）
}

// 上传响应消息
message PROTOLusp_UploadResponse {
    PROTOLusp_UploadResultInfo result = 1; // 上传结果
    
}